presto-hive/src/main/java/com/facebook/presto/hive/security/SqlStandardAccessControl.java

```diff
@@ -376,6 +377,20 @@ public void checkCanShowGrants(ConnectorTransactionHandle transaction, Identity
         return grantInfos;
     }
 
+    @Override
+    public void checkCanShowRoles(ConnectorTransactionHandle transactionHandle, Identity identity, String catalogName)
```

Comment:
Well, we have to work with how access control checks are currently implemented. This implies:
1) If we implement a method `checkCanShowCatalogs`, this method will be adding an access control check on *who* can run the `SHOW CATALOGS` command, which is not relevant for file based catalog access control. So this won't be the right place to deny catalog access for all other commands.
2) `filterCatalogs` can't throw an exception and hence this isn't the right place either.
3) Another option is to add a `checkCanAccessCatalog` method to SystemAccessControl, which would check catalog access, denying when necessary. But the caveat is that, such a `checkCanAccessCatalog` will have to be invoked explicitly in `AccessControlManager.java` for every `xxx` command. Thus, every `xxx` command, on which there needs to be an access control check either at a system level or at connector level, needs to have a corresponding `checkCanXxx`. This would ensure that the corresponding `checkCanXxx` method in `AccessControlManager` can invoke `checkCanAccessCatalog` method.